File: solutions/go/clock/1/clock.go
package clock

import "fmt"

// Define the Clock type here.
type Clock struct {
    hour int
    minute int
}

func New(h, m int) Clock {
    H := (h + m / 60) % 24
    for H < 0 {
        H += 24
    }
    M := m % 60
    for M < 0 {
        M += 60
        H -= 1
    }
    for H < 0 {
        H += 24
    }
    return Clock{hour: H, minute: M}
    
}

func (c Clock) Add(m int) Clock {
	total_minutes := c.minute + m;
    hour := (c.hour + total_minutes / 60) % 24;
    for hour < 0 {
        hour += 24;
    }
    minute := total_minutes % 60;
    for minute < 0 {
        minute += 60;
        hour -= 1;
    }
    for hour < 0 {
        hour += 24;
    }
    return Clock{hour: hour, minute: minute}
}

func (c Clock) Subtract(m int) Clock {
	totalMinutes := c.minute - m
    hour := (c.hour + totalMinutes/60) % 24
    // Обработка отрицательных значений
    for hour < 0 {
        hour += 24
    }
    minute := totalMinutes % 60
    // Если минуты отрицательные, то корректируем
    if minute < 0 {
        minute += 60
        hour -= 1
    }
    // Обработка отрицательных значений часов
    for hour < 0 {
        hour += 24
    }
    return Clock{hour: hour, minute: minute}
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

File: solutions/c/clock/1/clock.c
#include "clock.h"
#include <string.h>
#include <stdio.h>
#include <stdbool.h>

// typedef struct {
//    char text[MAX_STR_LEN];
// } clock_t;

clock_t clock_create(int hour, int minute) {
    clock_t result;
    while (minute < 0) {
        minute += 60;
        hour--;
    }
    int h = (hour + minute / 60) % 24;
    int m = minute % 60;
    while (h < 0) {
        h += 24;
    }
    
    snprintf(result.text, MAX_STR_LEN, "%02d:%02d", h, m);
    return result;
}
clock_t clock_add(clock_t clock, int minute_add) {
    int h, m;
    sscanf(clock.text, "%d:%d", &h, &m);
    int total_m = m + minute_add;
    h = (h + total_m / 60) % 24;
    while (h < 0) {
        h += 24;
    }
    m = total_m % 60;
    while (m < 0) {
        m += 60;
        h -= 1;
    }
    while (h < 0) {
        h += 24;
    }
    clock_t result;
    snprintf(result.text, MAX_STR_LEN, "%02d:%02d", h, m);
    return result;
}
clock_t clock_subtract(clock_t clock, int minute_subtract) {
    int h, m;
    sscanf(clock.text, "%d:%d", &h, &m);
    int total_m = m - minute_subtract;
    h = (h + total_m / 60) % 24;
    
    while (h < 0) {
        h += 24;
    }
    m = total_m % 60;
    
    if (m < 0) {
        m += 60;
        h -= 1;
    }
    
    while (h < 0) {
        h += 24;
    }

    clock_t result;
    snprintf(result.text, MAX_STR_LEN, "%02d:%02d", h, m);
    return result;
}
bool clock_is_equal(clock_t a, clock_t b) {
    return strcmp(a.text, b.text) == 0;
}